AndroidAdvance/app/src/main/java/com/example/androidadvance/MainActivity.kt

```diff
@@ -0,0 +1,52 @@
+package com.example.androidadvance
+import android.Manifest
+import android.os.Build
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import androidx.annotation.RequiresApi
+import androidx.viewpager.widget.ViewPager
+import com.example.androidadvance.adapter.FragmentAdapter
+import com.google.android.material.tabs.TabLayout
+import com.google.android.material.tabs.TabLayout.OnTabSelectedListener
+import kotlinx.android.synthetic.main.activity_main.*
+
+class MainActivity : AppCompatActivity() {
+
+    private lateinit var mTabLayout: TabLayout
+    private lateinit var mViewPager: ViewPager
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_main)
+
+        mTabLayout = tabLayout
```

Comment:
use id of view in xml instead define new one like this. (Should be use tabLayout instead mTabLayout)